controllers/topicPage/TopicPage.js

```diff
@@ -62,26 +96,17 @@ class TopicPage {
         offset,
         userId: req.userId
       };
-      let fetch = 0;
-      while (data.length < limit && fetch < MAX_FEED_FETCH_LIMIT) {
-        if (fetch !== 0) {
-          payload.offset = parseInt(payload.offset) + parseInt(limit);
-        }
-        let newTopicPagesWithBlock = await this.getFeedTopicPages(payload);
-
-        for (let post of newTopicPagesWithBlock) {
-          if (post?.is_hide) continue;
-          data.push(await this.modifiedFeed(req.userId, post));
-        }
-        fetch += 1;
-      }
+
+      let feeds = await this.getValidActivity(req.userId, payload, limit);
 
       res.status(200).json({
         code: 200,
         status: 'success',
         message: 'Success get topic pages',
-        data: data || [],
-        offset: parseInt(offset)
+        data: feeds.data || [],
+        offset: feeds.noActivity
+          ? parseInt(payload.offset)
+          : parseInt(payload.offset) + parseInt(limit)
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The response object has been updated to include the `feeds.data` and the offset. This is a good change as it provides more information in the response. However, the error handling in the catch block could be improved. Instead of just logging the error, it would be better to send a response with an appropriate status code and message.

```diff
-     } catch (error) {
-       console.log(error);
+     } catch (error) {
+       res.status(500).json({
+         code: 500,
+         status: 'error',
+         message: 'An error occurred while getting topic pages',
+         error: error.message
+       });
      }
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->